internal/common/errors/error_test.go

```diff
@@ -0,0 +1,73 @@
+// © 2023 Snyk Limited All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/snyk/container-cli/internal/common/errors"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Error_GivenErrorWithUserMessageAndWithoutWorkflowError_ShouldReturnUserMessageOnly(t *testing.T) {
+	err := "test error"
+	userMessage := "test user message"
+	errorCode := 42
+	workflowErrorCode := ""
+
+	unit := errors.NewContainerExtensionError(fmt.Errorf(err), userMessage, errorCode, workflowErrorCode)
+	result := unit.Error()
+
+	require.Equal(t, userMessage, result)
+}
+
+func Test_Error_GivenErrorWithUserMessageAndWorkflowErrorCodeAndErrorCode_ShouldReturnUserMessageAndAppendWorkflowErrorCodeAndErrorCode(t *testing.T) {
+	err := "test error"
+	userMessage := "test user message"
+	errorCode := 42
+	workflowErrorCode := "A"
+
+	expectedErrorMessage := fmt.Sprintf("%s [ERR#%s%d]", userMessage, workflowErrorCode, errorCode)
+
+	unit := errors.NewContainerExtensionError(fmt.Errorf(err), userMessage, errorCode, workflowErrorCode)
+	result := unit.Error()
+
+	require.Equal(t, expectedErrorMessage, result)
```

Comment:
nit: I think it would read much better if we hardcoded the expected error:

```suggestion
	require.Equal(t, "test user message [ERRA42]", result)
```

Reply:
Otherwise I have no idea what's the expectation (`"%s [ERR#%s%d]"` is not very human-readable)